macros/src/from_macro.rs

```diff
@@ -123,12 +124,50 @@ struct Register {
     byte_order: Option<ByteOrder>,
     reset_value: Option<ResetValue>,
     description: Option<String>,
+    cfg_attributes: Vec<syn::Attribute>,
     fields: Punctuated<Field, syn::Token![,]>,
 }
 
 impl Register {
-    fn parse(input: syn::parse::ParseStream, attributes: Vec<Attribute>) -> syn::Result<Self> {
-        let description = doc_string_from_attrs(&attributes)?;
+    fn parse(input: syn::parse::ParseStream, attributes: Vec<syn::Attribute>) -> syn::Result<Self> {
+        let (description, cfg_attributes) = {
+            let mut description = String::new();
+            let mut cfg_attributes = Vec::new();
+
+            for attr in attributes {
+                let path = attr.meta.path().require_ident()?.to_string();
+                match path.as_str() {
+                    "doc" => {
+                        let name_value = attr.meta.require_name_value()?;
+                        if let syn::Expr::Lit(syn::ExprLit {
+                            lit: syn::Lit::Str(value),
+                            ..
+                        }) = &name_value.value
+                        {
+                            description += &value.value();
+                            continue;
+                        }
+                    }
+                    "cfg" => {
+                        cfg_attributes.push(attr);
+                        continue;
+                    }
+                    _ => {}
+                }
+                return Err(syn::Error::new(
+                    attr.meta.path().span(),
+                    format!("Attribute type `{path}` not supported in this usecase"),
+                ));
+            }
+
+            let description = if description.is_empty() {
+                None
+            } else {
+                Some(description)
+            };
+
+            (description, cfg_attributes)
+        };
```

Comment:
Let's make this a function just like `doc_string_from_attrs` was. Then we can reuse it in other parts.